### PR TITLE
Adding NXP imx6sll-evk, imx7ulp-evk, imx8mm-ddr4-evk and imx8mp-evk t…

### PR DESCRIPTION
**Pull request by suram-nxp**
Adding NXP's 4 more boards that could not be pushed as part of Pull Request (https://github.com/kernelci/kernelci-core/pull/412).

### CODE REVIEW

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:655]: This board doesn't appear to be in linux-next, from `next-20200720`:
```
$ ls arch/arm64/boot/dts/freescale/*ddr4*
arch/arm64/boot/dts/freescale/imx8mn-ddr4-evk.dts
```
Only the `imx8mn-ddr4-evk` is there. Maybe `imx8mn-ddr4-evk` should be dropped for now?  It's not going to be tested in KernelCI until it's at least in linux-next.  Is it on its way to a maintainer branch?

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:667]:
```suggestion
    arch: arm64
    dtb: 'freescale/imx8mp-evk.dtb'
```

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:662]:
```suggestion
    arch: arm64
    dtb: 'freescale/imx8mn-ddr4-evk.dtb'
```

**Comment by suram-nxp** [comment]
Thank you very much for helping in identifying the issue.  
Your suggestions were discussed and accepted by the board owners.
I have removed i.mx8mm board from kernelci as well.
I hope the tests will pass now.

**Comment by suram-nxp** [comment]
Hello gctucker,
i.mx8 issues are now resolved.  Thank you very much for the support provided in identifying the issue.
Jobs are still not getting scheduled for i.mx6sll-evk and imx7ulp-evk boards.  I have ensured that
1. device type file name matches with upstream dtb file name
2. device type file is already available on LAVA
Ideally, there should not be any problem as the jobs for other i.mx7 and 6 boards are getting triggered.

I request your kind support in identifying this issue too.
Thank you very much for the support.

**Comment by gctucker** [comment]
Thanks, sorry I've been very busy with shifting the production Jenkins instance.  I'll do another review of this PR on Monday, the changes look fine but maybe there's a small glitch somewhere.

**Comment by khilman** [comment]
Hi @suram-nxp,

I tried to test this today, and in my local test `kci_test get_lab_info` is getting a `502 Proxy Error` when trying to get the lab info.  I double checked the tokens, and that all looks right. 

When logging into the NXP LAVA server manually, it seems very slow, so maybe there's some timeout issue going on when accessing LAVA?

**Comment by khilman** [comment]
Also, please remove the `kselftest` test-plan from any new boards.  That test-plan is not working as-is, and is being reworked in current PRs.  Supporting kselftest will now require NFSroot support, so that will be available on a smaller number of boards.

**Comment by khilman** [comment]
Here's at least part of the problem for imx6sll, that SoC family is not enabled in the upstream defconfig.  Here's the `.config` from mainline today:
```
#
# Cortex-A platforms
#
CONFIG_SOC_IMX5=y
CONFIG_SOC_IMX50=y
CONFIG_SOC_IMX51=y
CONFIG_SOC_IMX53=y
CONFIG_SOC_IMX6=y
CONFIG_SOC_IMX6Q=y
CONFIG_SOC_IMX6SL=y
# CONFIG_SOC_IMX6SLL is not set
CONFIG_SOC_IMX6SX=y
CONFIG_SOC_IMX6UL=y
CONFIG_SOC_LS1021A=y
```

and same for imx7ulp:
```
#
# Cortex-A/Cortex-M asymmetric multiprocessing platforms
#
CONFIG_SOC_IMX7D_CA7=y
CONFIG_SOC_IMX7D=y
# CONFIG_SOC_IMX7ULP is not set
```

**Comment by khilman** [comment]
It looks like these are enabled in `imx_v6_v7_defconfig` but not in the default `multi_v7_defconfig`.  I'm not sure if we build the `imx_v6_v7_defconfig` on staging, so that might explain why those boards are not tested in staging.

I did a quick test locally, using `imx_v6_v7_defconfig` and see LAVA jobs created for imx6ll and imx7ulp:
https://lavalab.nxp.com/scheduler/alljobs?length=25&search=mainline-master-v5.8-rc7-arm-imx_v6_v7_defconfig#table

File: test-configs.yaml
Comment (khilman) [inline test-configs.yaml:1675]: Please drop the kselftest from these new test-plans.  We're reworking kselftest, and going forward, kselftest will require boards to support NFSroot.

**Comment by khilman** [comment]
My locally-generated jobs (created using kci_test from staging branch) working in lab-nxp:

imx6sll: https://lavalab.nxp.com/scheduler/job/49068#bottom
imx7ulp: https://lavalab.nxp.com/scheduler/job/49069#bottom

So after the kselftest test-plan is dropped from the new devices (as mentioned in an earlier comment), this PR looks good to me.

**Comment by gctucker** [comment]
> When logging into the NXP LAVA server manually, it seems very slow, so maybe there's some timeout issue going on when accessing LAVA?

I believe this is a bug in LAVA, we're seeing this as well with the Collabora lab and others tend to be slow too looking at the KernelCI Jenkins logs.  As long as it eventually returns the list of devices then it shouldn't be a real issue, but it would be better to fix the root cause.

**Comment by gctucker** [comment]
> So after the kselftest test-plan is dropped from the new devices (as mentioned in an earlier comment), this PR looks good to me.

FYI I've set filters in `lab-configs.yaml` to not run `kselftest` in any lab as there are tens of `test_config` entries with it.  But yes, let's not add some more to set the correct expectations as this is not working yet.  A PR to remove all the `kselftest` mentions in `test-configs.yaml` would probably help.

**Comment by aishengdong** [comment]
> Here's at least part of the problem for imx6sll, that SoC family is not enabled in the upstream defconfig. Here's the `.config` from mainline today:
> 
> ```
> #
> # Cortex-A platforms
> #
> CONFIG_SOC_IMX5=y
> CONFIG_SOC_IMX50=y
> CONFIG_SOC_IMX51=y
> CONFIG_SOC_IMX53=y
> CONFIG_SOC_IMX6=y
> CONFIG_SOC_IMX6Q=y
> CONFIG_SOC_IMX6SL=y
> # CONFIG_SOC_IMX6SLL is not set
> CONFIG_SOC_IMX6SX=y
> CONFIG_SOC_IMX6UL=y
> CONFIG_SOC_LS1021A=y
> ```
> 
> and same for imx7ulp:
> 
> ```
> #
> # Cortex-A/Cortex-M asymmetric multiprocessing platforms
> #
> CONFIG_SOC_IMX7D_CA7=y
> CONFIG_SOC_IMX7D=y
> # CONFIG_SOC_IMX7ULP is not set
> ```
> 
> As a result, the DTBs for these platforms are not built (c.f. `arch/arm/boot/dts/Makefile`), and when the DTBs are not build, KernelCI will not generate LAVA jobs.

Thanks for letting us know.
I just send two patches to enable those two SoCs in multi_v7_defconfig.
https://www.spinics.net/lists/arm-kernel/msg826181.html

Regards
Aisheng

**Comment by suram-nxp** [comment]
@gctucker, 
If I understand correctly, kselftests are not triggered even if they are part of test-configs.yaml file and hence should not impact the trigger.
Since the config is not enabled in in multi_v7_defconfig, the build did not happen and hence jobs are not triggered automatically.
As @aishengdong  has submitted, once it is merged the jobs should be triggered.

Once it happens and the PR is merged, I will raise a separate PR to remove kselftests from test-configs.yaml so that the file has only the test plans that can be executed now.

**Comment by gctucker** [comment]
Sounds good.  It's also possible to have an extra build on staging with `imx_v6_v7_defconfig` just to prove the devices and the config is correct, I can get one scheduled manually later today.

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:625]: Tested OK on staging: https://staging.kernelci.org/test/plan/id/5f330f301e9528386e47b0d6/

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:650]: Tested OK on staging: https://staging.kernelci.org/test/plan/id/5f330f66dd24528e8247b0d0/

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:1675]:
```suggestion
```

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:655]: Tested OK on staging: https://staging.kernelci.org/test/plan/id/5f32c7a207c8adc0ca47b0db/

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:1700]: Same issue, please drop `kselftest`.
```suggestion
```

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:1710]:
```suggestion
```

File: test-configs.yaml
Comment (gctucker) [inline test-configs.yaml:661]: This one has run but it failed to reach a login prompt, it seemed stuck waiting for the network interface:
https://staging.kernelci.org/test/plan/id/5f32c8abe9e0d5d52e47b0e3/
https://storage.staging.kernelci.org/kernelci/staging.kernelci.org/staging-20200811.2/arm64/defconfig/gcc-8/lab-nxp/baseline-imx8mp-evk.html

**Comment by suram-nxp** [comment]
Thank you very much @gctucker for the feedback.  At the moment I am working from home due to current situation. Please allow me couple of days and I will go to office to check the board status of i.mx8mp-evk and imx8mn-evk. 
I will also add the list of device-type files in LAVA.
Also, I will update the PR and remove all kselftests for NXP boards.

**Comment by suram-nxp** [comment]
1. 

1. i.mx8mp-evk board issue - Resolved the issue.  Issue is with cable connected to wrong ethernet port.
2. Device type files on LAVA
imx6sll-evk : https://git.lavasoftware.org/lava/lava/-/blob/master/etc/dispatcher-config/device-types/imx6sll-evk.jinja2
imx7ulp-evk : https://git.lavasoftware.org/lava/lava/-/blob/master/etc/dispatcher-config/device-types/imx7ulp-evk.jinja2
imx8mm-ddr4-evk : https://git.lavasoftware.org/lava/lava/-/blob/master/etc/dispatcher-config/device-types/imx8mm-ddr4-evk.jinja2
imx8mp-evk : https://git.lavasoftware.org/lava/lava/-/blob/master/etc/dispatcher-config/device-types/imx8mp-evk.jinja2
3. Removed kself tests for all NXP's i.MX boards
4. i.mx8mm-ddr4-evk was missing.  Added the board.

This took time as I could not go to office for couple of weeks due to current situation.
Please let me know if anything else is required.

File: test-configs.yaml
Comment (gctucker) [reply test-configs.yaml:633]: Tested OK on staging now: https://staging.kernelci.org/test/plan/id/5f4f57b549d566666d39b1e8/

**Review by gctucker — APPROVED** [review]
OK thanks for addressing all the issues and getting the boards online.  I believe it's all ready to be merged now.

**Pull request merged**
